src/main/java/org/vanilladb/core/storage/metadata/ViewMgr.java

```diff
@@ -86,4 +90,20 @@ public String getViewDef(String vName, Transaction tx) {
 		rf.close();
 		return result;
 	}
+
+	public Collection<String> getViewNameByTable(String tblName, Transaction tx) {
+		Collection<String> result = new LinkedList();
+
+		TableInfo ti = tblMgr.getTableInfo(VCAT, tx);
+		RecordFile rf = ti.open(tx, true);
+		rf.beforeFirst();
+		while (rf.next()) {
+			Parser parser = new Parser((String) rf.getVal(VCAT_VDEF).asJavaVal());
```

Comment:
It is an interesting idea to use `Parser` here, but this may make the storage engine depend on the query engine, which violates separation of concerns in our design. However, I have not come out a better idea yet. So, we can accept this for now.
Please add the following comment in front of the code:
```
// XXX: this makes the storage engine depend on the query engine.
// We may have to come out a better method.
```